tooling/benchmark.py

```diff
@@ -1,110 +1,72 @@
-            websocket_duration = (
-                time.time() - start_time - TRANSCRIPTION_WEBSOCKET_TIMEOUT
-            )
-        if settings.debug:
-            print(f"Expected: {expected}")
-
-        results["key"] += [get_file_name(filepath, settings.scale)]
-        for val in [
-            (rest_transcription, rest_duration, "rest"),
-            (websocket_transcription, websocket_duration, "websocket"),
-        ]:
-            transcription, duration, method_key = val
-
-            if settings.use_partials_for_websocket and method_key == "websocket":
-                results[f"wer_{method_key}"] += [websocket_partial_result]
-                results[f"duration_{method_key}"] += [duration]
-                continue
+            websocket_result = transcribe_file_websocket(
+                filepath, settings.overwrite_api_key, settings.scale, settings.debug)
+        key = get_file_name(filepath, settings.scale)
+        export_file = os.path.join(os.getcwd(), "export", f"{key[:-4]}.json")
 
-            if transcription is None or duration is None:
-                results[f"wer_{method_key}"] += [None]
-                results[f"duration_{method_key}"] += [None]
-                continue
-
-            results[f"duration_{method_key}"] += [duration]
-
-            # This can only happen if backend or benchmark encounter an error
-            # However it WOULD terminate the entire benchmarking process
-            if len(transcription) == 0:
-                results[f"wer_{method_key}"] += [ERR_CODE]
-                continue
-
-            transcription = TRANSFORM_DEFAULT(transcription)
-            if settings.debug:
-                print(f"Received({method_key}): {transcription}")
+        grouped_result = BenchmarkResult(
+            filename=key , 
+            rest=rest_result, 
+            websocket=websocket_result, 
+            scale=settings.scale,
+            expected_transcription=get_corresponding_transcript(export_file, settings.scale)
+        )
 
-            curr_wer = jiwer.wer(" ".join(transcription), " ".join(expected))
-            results[f"wer_{method_key}"] += [curr_wer]
+        with open(export_file, "w+") as f:
+            json.dump(asdict(grouped_result),f)
 
     if settings.scale_percentage != "100":
         print(
-            f"Premature termination after {int(limit)} (dataset total length: {len(audio_files)}) evaluated audios. This was caused by the set limit percentage of {settings.scale_percentage}"
+            f"Premature termination after {limit} (dataset total length: {len(audio_files)}) evaluated audios. This was caused by the set limit percentage of {settings.scale_percentage}"
         )
 
-    render_table(results, export=settings.export_markdown)
+def benchmark(settings):
+    if not settings.skip_fetch:
+        perform_fetches(settings)
 
+    res = eval_export_dir()
+    print(f"{'-'*5} DATA {'-'*5}")
```

Comment:
kinda at this point, its just less readable instead of shorter ... and smooth xD
```suggestion
    print("----- DATA -----")
```